heron/common/src/java/com/twitter/heron/common/basics/FileUtils.java

```diff
@@ -116,4 +116,21 @@ public static String getBaseName(String file) {
     return new File(file).getName();
   }
 
+  public static boolean deleteDir(String dir) {
+    return deleteDir(new File(dir));
+  }
+
+  public static boolean deleteDir(File dir) {
+    if (dir.isDirectory()) {
+      String[] children = dir.list();
+      for (int i = 0; i < children.length; i++) {
+        boolean success = deleteDir(new File(dir, children[i]));
+        if (!success) {
+          return false;
```

Comment:
The changes are on the ckptmgr side of things. I don't want to change in the FileUtils layer yet because all other calls there are all returning status as return value(as opposed to throwing exceptions).